src/Layout.ts

```diff
@@ -108,6 +120,78 @@ export class LayoutSystem
      */
     resize(parentWidth: number, parentHeight: number)
     {
+        let finalStyles = { ...this.options?.styles };
+
+        if (this.dpr !== 'portrait' && this.options?.styles?.portrait && parentHeight >= parentWidth)
+        {
+            this.dpr = 'portrait';
+            finalStyles = { ...finalStyles, ...this.options.styles.portrait };
+
+            this.setStyles(finalStyles);
+        }
+
+        if (this.dpr !== 'landscape' && this.options?.styles?.landscape && parentHeight < parentWidth)
+        {
+            this.dpr = 'landscape';
+            finalStyles = { ...finalStyles, ...this.options.styles.landscape };
+
+            this.setStyles(finalStyles);
+        }
+
+        if (this.options?.styles?.max)
+        {
+            if (this.options.styles.max.height)
+            {
+                for (const [key, value] of Object.entries(this.options.styles.max.height))
+                {
+                    if (parentHeight <= parseInt(key, 10))
+                    {
+                        finalStyles = { ...finalStyles, ...value };
+                    }
+                }
+            }
+
+            if (this.options.styles.max.width)
+            {
+                for (const [key, value] of Object.entries(this.options.styles.max.width))
+                {
+                    if (parentWidth <= parseInt(key, 10))
+                    {
+                        finalStyles = { ...finalStyles, ...value };
+                    }
+                }
+            }
+
+            this.setStyles(finalStyles);
+        }
+
+        if (this.options?.styles?.min)
+        {
+            if (this.options.styles.min.height)
+            {
+                for (const [key, value] of Object.entries(this.options.styles.min.height))
+                {
+                    if (parentHeight >= parseInt(key, 10))
+                    {
+                        finalStyles = { ...finalStyles, ...value };
+                    }
+                }
+            }
+
+            if (this.options.styles.min.width)
+            {
+                for (const [key, value] of Object.entries(this.options.styles.min.width))
+                {
+                    if (parentWidth >= parseInt(key, 10))
+                    {
+                        finalStyles = { ...finalStyles, ...value };
+                    }
+                }
+            }
+
+            this.setStyles(finalStyles);
+        }
+
```

Comment:
Here is where all the logic of this feature lives.
It tries to detect whether conditional styles are set in props and should be applied, then it mixes a conditional styles with main styles and applies them.